### DISCUSSION TITLE
Does Kratos have a gRPC API?

### DISCUSSION THREAD

**Discussion by san4d** [ory/kratos#2775] [Q&A]
I've just started playing around with Ory Cloud and Kratos. I see mention of a gRPC API in the [crate documentation](https://crates.io/crates/ory-kratos-client), but I cannot find the protobuf definitions. I've checked some generated client repositories ([example](https://github.com/ory/kratos-client-rust)), but I only see the [OpenAPI Generator](https://openapi-generator.tech/) types. I also do not see mention of protobufs or gRPC in the [sdk repo](https://github.com/ory/sdk).

**Comment by vinckr** [comment]
Kratos has a [REST API](https://www.ory.sh/docs/kratos/reference/api) only at the time. I am not sure if gRPC for Kratos was discussed in the past, but I did not find much in a quick search.

Ory Keto does have a [gRPC API](https://www.ory.sh/docs/keto/reference/proto-api), you can use it for authorization use cases.

**Comment by allenkaplan** [comment]
As far as I know, no -- would be a nice to have for better interoperability
